cli/cycpp.py

```diff
@@ -1568,10 +1578,10 @@ def impl(self, ind="  "):
 
             if key in BUFFERS:  # buffer state, skip
                 continue
-            if 'derived_init' in info:  # derived state, skip
+            if 'internal' in info:  # derived state, skip
                 continue
 
-            opt = True if 'default' in info else False
+            opt = True if 'default' in info or 'derived_init' in info else False
```

Comment:
If we don't change this line, users will be required to put a value in input files for all derived init vars - that would be very bad IMO.  The  primary use case for derived init is to allow the dev to specify a special sort of default value so the user doesn't have to manually.